libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/LocationObserver.kt

```diff
@@ -4,6 +4,8 @@ import android.location.Location
 
 /**
  * An interface which enables listening to location updates
+ *
+ * @see [LocationMatcherResult]
  */
 interface LocationObserver {
```

Comment:
Do we still want to change the name of the `interface`? https://github.com/mapbox/mapbox-navigation-android/issues/4814#issuecomment-924601371 cc @LukasPaczos @abhishek1508

Reply:
I think `LocationMatcherResultObserver` would be a bit misleading, because apart from `LocationMatcherResult`s the interface allows to observe raw `Location`s.